src/disco/topo/fd_topo.c

```diff
@@ -252,6 +252,18 @@ fd_topo_gigantic_page_cnt( fd_topo_t * topo ) {
   return result;
 }
 
+FD_FN_PURE ulong
+fd_topo_mounted_huge_page_cnt( fd_topo_t * topo ) {
+  ulong result = 0UL;
+  for( ulong i=0UL; i<topo->wksp_cnt; i++ ) {
+    if( FD_LIKELY( topo->workspaces[ i ].page_sz == FD_SHMEM_HUGE_PAGE_SZ ) ) {
+      result += topo->workspaces[ i ].page_cnt;
+    }
+  }
+
+  return result;
+}
+
 FD_FN_PURE ulong
 fd_topo_huge_page_cnt( fd_topo_t * topo ) {
   ulong result = 0UL;
```

Comment:
You can replace the loop below with a call to `fd_topo_mounted_huge_page_cnt`

Reply:
Or what about making this one function with a bool flag whether to include extra (non-wksp) pages or not? `mounted` doesn't seem like the right word to describe these pages to me